src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -145,8 +145,8 @@ State Propagator::calculateStateAt(const State& aState, const Instant& anInstant
 
     const State state = {
         aState.accessInstant(),
-        extractCoordinatesFromState(aState.inFrame(Propagator::IntegrationFrameSPtr)),
-        aState.accessFrame(),
+        aState.inFrame(Propagator::IntegrationFrameSPtr).extractCoordinates(this->coordinatesBrokerSPtr_->getSubsets()),
+        Propagator::IntegrationFrameSPtr,
```

Comment:
Someone double check me here. Since we're always converting the coordinates to the Propagator Frame, shouldn't we always be setting the State to that Frame as well?

Reply:
That's a good point. I think it was working out cause we convert everything to further up the chain. Great catch!